librz/core/cmd/cmd_print.c

```diff
@@ -1716,48 +1716,94 @@ static void pr_bb(RzCore *core, RzAnalysisFunction *fcn, RzAnalysisBlock *b, boo
 	}
 }
 
-static void disasm_until_ret(RzCore *core, ut64 addr, int limit, RzOutputMode mode) {
-	int p = 0;
+static void handle_default_disasm_print_mode(const RzCore *core, const ut64 addr, const bool ret_val,
+	const char *m_intr, RzStrBuf *buf) {
 	const bool show_color = rz_config_get_i(core->config, "scr.color");
+
+	if (show_color) {
+		const char *offsetColor = rz_cons_singleton()->context->pal.offset;
+		if (!ret_val) {
+			rz_cons_printf("%s0x%08" PFMT64x "" Color_RESET "  %10s %s\n",
+				offsetColor, addr, "", m_intr);
+		} else {
+			char *str_buf = rz_str_newf("%s0x%08" PFMT64x "" Color_RESET "  %10s %s\n",
+				offsetColor, addr, "", m_intr);
+			rz_strbuf_append(buf, str_buf);
+			free(str_buf);
+		}
+	} else {
+		if (!ret_val) {
+			rz_cons_printf("0x%08" PFMT64x "  %10s %s\n", addr, "", m_intr);
+		} else {
+			char *str_buf = rz_str_newf("0x%08" PFMT64x "  %10s %s\n", addr, "", m_intr);
+			rz_strbuf_append(buf, str_buf);
+			free(str_buf);
+		}
+	}
+}
+
+static void disasm_print_ret(const RzCore *core, const RzOutputMode mode, const char *m_intr, const ut64 addr,
+	const bool ret_val, RzStrBuf *buf) {
+	switch (mode) {
+	case RZ_OUTPUT_MODE_QUIET:
+		if (!ret_val) {
+			rz_cons_printf("%s\n", m_intr);
+		} else {
+			rz_strbuf_append(buf, m_intr);
+		}
+		break;
+	default:
+		handle_default_disasm_print_mode(core, addr, ret_val, m_intr, buf);
+		break;
+	}
+}
+
+/**
+ * \brief Disassemble from \p addr until \p limit no of instructions and
+ * prints or returns the disassembled string in \p buf
+ *
+ * \param core Pointer to the RzCore
+ * \param addr Address to start disassembling
+ * \param limit Maximum number of instructions to disassemble
+ * \param mode The mode in which data has to be printed/crafted
+ * \param ret_val Flag to indicate if the disassembled string has to be stored in \p buf
+ * \param buf Pointer to the RzStrBuf to store the disassembled string
+ *
+ * \return True if there are no errors during disassembling and crafting the response else False
+ */
+RZ_API bool rz_core_disasm_until_ret(RZ_NONNULL RzCore *core, ut64 addr, const int limit,
+	const RzOutputMode mode, const bool ret_val, RZ_NONNULL RZ_BORROW RzStrBuf *buf) {
```

Comment:
```suggestion
	const RzOutputMode mode, const bool ret_val, RZ_NONNULL RZ_OUT RzStrBuf *buf) {
```

Is a little more descriptive.